presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java

```diff
@@ -623,19 +629,19 @@ public void close()
             List<WriterEncryptionGroup> writerEncryptionGroups = dwrfWriterEncryption.get().getWriterEncryptionGroups();
             for (int i = 0; i < writerEncryptionGroups.size(); i++) {
                 WriterEncryptionGroup group = writerEncryptionGroups.get(i);
-                List<byte[]> columnStatistics = encryptedStats.get(i)
-                        .stream()
-                        .map(DwrfMetadataWriter::toColumnStatistics)
-                        .map(AbstractMessageLite::toByteArray)
-                        .collect(toImmutableList());
-
                 DwrfProto.FileStatistics fileStatistics = toFileStatistics(encryptedStats.get(i));
-                byte[] serializedStats = fileStatistics.toByteArray();
+                DwrfDataEncryptor dwrfDataEncryptor = dwrfEncryptionInfo.getEncryptorByGroupId(i);
+                OrcOutputBuffer buffer = new OrcOutputBuffer(compression, Optional.of(dwrfDataEncryptor), maxCompressionBufferSize);
+                fileStatistics.writeTo(buffer);
```

Comment:
do we have unit tests for stats of nested types?

Reply:
We don't have any fileStatistics tests (Presto doesn't use filestatistics for dwrf, so none of our tests are testing the functionality). Let me think about how to write some kind of test though and I'll add it as a follow on